### PR TITLE
Audit Fixes

### PR DESCRIPTION
**Pull request by leomassazza**

### CODE REVIEW

File: contracts/PerpsV2Market.sol [hunk 1]
```diff
@@ -39,34 +39,50 @@ import "./interfaces/IPerpsV2Market.sol";
  *                                  the base asset, these settings determine the behaviour of each market.
  *                                  See that contract for descriptions of the meanings of each setting.
  *
+ * Proxy/Implementations/State Contracts architecture diagram
+ *
+ *                 ┌-> PerpsV2Market                 ─┐
+ *                 ├-> PerpsV2MarketViews            -┤
+ *   ProxyPerpsV2 ─┼-> PerpsV2MarketLiquidate        ─┼─> PerpsV2MarketState -> PerpsV2MarketStateLegacyR1
```
Comment (davidvuong) [inline contracts/PerpsV2Market.sol:46]: ❤️ I love this diagram

File: contracts/FuturesMarketManager.sol [hunk 2]
```diff
@@ -441,10 +441,8 @@ contract FuturesMarketManager is Owned, MixinResolver, IFuturesMarketManager {
      */
     function addEndorsedAddresses(address[] calldata addresses) external onlyOwner {
         for (uint i = 0; i < addresses.length; i++) {
-            if (!_endorsedAddresses.contains(addresses[i])) {
-                _endorsedAddresses.add(addresses[i]);
-                emit EndorsedAddressAdded(addresses[i]);
-            }
+            _endorsedAddresses.add(addresses[i]);
```
Comment (davidvuong) [inline contracts/FuturesMarketManager.sol:444]: Oh good catch. So every call just wipes it from index 0. I guess the problem here is if `addresses` is shorter than `_endorsedAddresses`, we can end up with lingering prior addresses.

File: contracts/PerpsV2MarketDelayedExecution.sol [hunk 3]
```diff
@@ -308,19 +308,21 @@ contract PerpsV2MarketDelayedExecution is IPerpsV2MarketDelayedExecution, PerpsV
...
-            account,
-            position.margin,
-            position.size,
-            0,
-            fillPrice,
-            fundingIndex,
-            0,
-            marketState.marketSkew()
-        );
+        if (toRefund > 0) {
```
Comment (davidvuong) [inline contracts/PerpsV2MarketDelayedExecution.sol:311]: Strange I thought I had added this before but I could be mistaken 🤔

**Review by davidvuong — APPROVED** [review]
LGTM 👍

File: contracts/FuturesMarketManager.sol (hunk 2 above)
Comment (leomassazza) [reply contracts/FuturesMarketManager.sol:444]: is not a replacement, it adds all the addresses in the input array

File: contracts/PerpsV2MarketDelayedExecution.sol (hunk 3 above)
Comment (leomassazza) [reply contracts/PerpsV2MarketDelayedExecution.sol:311]: me too... maybe a merge conflict?

**Pull request merged**